x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/PredictionFieldType.java

```diff
@@ -57,6 +57,19 @@ public Object transformPredictedValue(Double value, String stringRep) {
                 }
                 return areClose(value, 1.0D);
             case NUMBER:
+                if (stringRep == null) {
+                    return value;
+                }
+                // Quick check to verify that the string rep is LIKELY a number
```

Comment:
Two questions here:
1. What do we do with an empty string? Currently it would fail with `IndexOutOfBounds`, right?
2. Why do we prioritize `stringRep` over `value` even in a number case?

Reply:
1. Good question, I am not sure what to do in this case....Probably return underlying value.
2. Please read the PR description. When creating a model in analytics, it is possible to have classification classes that are numeric, but not ordinal. These are treated as classification_labels.